.github/workflows/build-nuget.yml

```diff
@@ -0,0 +1,95 @@
+name: Build NuGet Package
+on:
```

Comment:
These triggers are nice!
Could you please also add the `workflow_dispatch` trigger, so one could rebuild the package by manually triggering it.

References:
* https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/
* https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow